accname-aam/accname-aam.html

```diff
@@ -323,7 +323,7 @@
                 <pre class="example highlight"><code>&lt;div role="checkbox" aria-checked="false"&gt;Flash the screen &lt;span role="textbox" aria-multiline="false"&gt; 5 &lt;/span&gt; times&lt;/div&gt;</code></pre>
               </details></div>
               </li>
-            <li id="step2F">Otherwise, if the <code>current node's</code> <a class="termref">role</a> allows <a class="specref" href="#namefromcontent">name from content</a>, or if the <code>current node</code> is referenced by <code>aria-labelledby</code>, <code>aria-describedby</code>, or is a native host language text alternative <a class="termref">element</a> (e.g. <code>label</code> in HTML):
+            <li id="step2F">Otherwise, if the <code>current node's</code> <a class="termref">role</a> allows <a class="specref" href="#namefromcontent">name from content</a>, or if the current node has no defined role and contains content, or if the <code>current node</code> is referenced by <code>aria-labelledby</code>, <code>aria-describedby</code>, or is a native host language text alternative <a class="termref">element</a> (e.g. <code>label</code> in HTML):
```

Comment:
@richschwer 
The original text has three conditions, such that if any one condition holds, then the content of the node is used.  The conditions are:
- role allows name from content,
- node is referenced by `aria-labelledby` or `aria-describedby`,
- node is a native labelling element (e.g. html `<label>`)

The second condition almost does the job.  If the `<em>` node were referenced via `aria-labelledby`, then its contents would be used.  The difference is that in Cynthia's example the `<em>` is a child of a `<label>` -- it's implicitly "referenced" via a `<label>` element.  I suggest extending the second condition to include native host language labelling elements:

- node is referenced by `aria-labelledby` or `aria-describedby`, or is a descendant node of a native host language text alternative element (e.g., `<label>` in HTML).

Technically, how various descendant html elements contribute within a `<label>` element should be specified in the html-aam or the html spec itself, if it currently isn't.